src/common/transformations/include/transformations/sdpa_to_paged_attention/position_ids_replacer.hpp

```diff
@@ -23,4 +24,21 @@ class ov::pass::PositionIDsReplacer : public ov::pass::MatcherPass {
 public:
     OPENVINO_RTTI("PositionIDsReplacer", "0");
     explicit PositionIDsReplacer(const Output<Node>& position_ids);
-};
\ No newline at end of file
+};
+
+/**
+ * @brief Qwen model has a specific feature in the model structure not to use position_ids input,
+ * this input is detached. The model expects data processing in order.
+ *
+ * To use this model in Continuous batching mode, we need to apply position_ids and
+ * use the corresponding rotary_emb_cos/rotary_emb_sin.
+ * For this, we replace
+ *      rotary_emb_cos/rotary_emb_sin -> Slice -> Slice
+ * With
+ *      rotary_emb_cos/rotary_emb_sin -> Slice -> Gather(by position_ids)
```

Comment:
... which enables applying RoPE for each token independently of their order in the input tensor.

Reply:
Why do we need to keep the first Slice?

Reply:
Probably we don't need, I will double check

Reply:
done